packages/gatsby/src/utils/page-data.ts

```diff
@@ -0,0 +1,79 @@
+import fs from "fs-extra"
+import path from "path"
+import { IGatsbyPage } from "../redux/types"
+import { store } from "../redux"
+
+interface IPageDataContext {
+  publicDir: string
+}
```

Comment:
I know this isn't the scope of this, but can we actually change this to not wrap these in an object? All of the invocations that are like `fn({ publicDir })` could and should just be `fn(publicDir)`. So this will require updating the references too